src/node_util.cc

```diff
@@ -67,6 +69,20 @@ static void GetConstructorName(
   args.GetReturnValue().Set(name);
 }
 
+static void GetExternalValue(
+    const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsExternal());
+  auto isolate = args.GetIsolate();
+  Local<External> external = args[0].As<External>();
+
+  void* ptr = external->Value();
+  char val[20];
+  snprintf(val, sizeof(val), "%p", ptr);
```

Comment:
`%p` can either format the pointer as `1234abcde` or `0x1234abcd`, and there's also potential inconsistency around NULL pointers. I'd cast the pointer to `uint64_t` and format it with `PRIx64` from `<cinttypes>`.

An alternative solution is to turn the pointer into a `BigInt` with `v8::BigInt::NewFromUnsigned()` and do the formatting in JS land with `value.toString(16)`.